frontend/src/pages/profile/components/user-activity/user-activity.tsx

```diff
@@ -0,0 +1,90 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { type UserActivityResponseDto } from '~/packages/users/users.js';
+
+import { ACTIVITY_BREAKPOINTS } from '../../libs/constants/constants.js';
+import {
+  getActivityBreakpoint,
+  getUniqueMonths,
+} from '../../libs/helpers/helpers.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  userActivity: UserActivityResponseDto[];
+};
+
+const UserActivity: React.FC<Properties> = ({ userActivity }) => {
+  const uniqueMonths: string[] = userActivity
```

Comment:
As Properties type says userActivity is an array of UserActivityResponseDto objects,, this means that it would always be passed as an array, but here you have ternary that checks if userActivity exists. Due to fact that empty array is a truthy value - there could be 2 cases:
1 - Properties type is not correct
2 - This check is not necessary